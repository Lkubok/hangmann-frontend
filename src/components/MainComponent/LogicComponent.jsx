import React, { Component } from "react";

class LogicComponent extends Component {
  state = { x: 0, y: 0 };

  handleMouseMove = event => {
    this.setState({
      x: event.clientX,
      y: event.clientY
    });
  };

  render() {
    return (
      <div onMouseMove={this.handleMouseMove} style={{ height: "100%" }}>
        {this.props.render(this.state)}
      </div>
    );
  }
}

export default LogicComponent;
