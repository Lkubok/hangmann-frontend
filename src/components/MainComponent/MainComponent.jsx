import React, { Component } from "react";

import LogicComponent from "./LogicComponent";

export default class App extends Component {
  render() {
    return (
      <div style={{ height: "100vh" }}>
        <LogicComponent
          render={({ x, y }) => (
            <h1 id="test">
              Coordinates: ({x}, {y})
            </h1>
          )}
        />
      </div>
    );
  }
}
