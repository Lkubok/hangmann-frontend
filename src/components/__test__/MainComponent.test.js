import React from "react";
import { mount } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import MainComponent from "../MainComponent/MainComponent";
Enzyme.configure({ adapter: new Adapter() });

let props;

describe("Search Box Component", () => {
  const spyFunction = jest.fn();
  beforeEach(() => {
    props = {
      filterQuery: null,
      pageSize: null,
      quotes: null,
      changeFilter: spyFunction,
      changeSortedQuotes: spyFunction,
      changePageSize: spyFunction
    };
  });

  it("Should match snapshot after mount", () => {
    const wrapper = mount(<MainComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should display proper coordinates (30)", () => {
    const wrapper = mount(<MainComponent {...props} />);
    const spyFunction = jest.fn();
    const event = {
      clientX: 30,
      clientY: 30
    };
    wrapper.instance().handleChange = spyFunction;
    wrapper.instance().forceUpdate();
    wrapper.find("#test").simulate("mouseMove", event);
    expect(wrapper.text()).toEqual("Coordinates: (30, 30)");
  });

  it("should display proper coordinates (40)", () => {
    const wrapper = mount(<MainComponent {...props} />);
    const spyFunction = jest.fn();
    const event = {
      clientX: 40,
      clientY: 40
    };
    wrapper.instance().handleChange = spyFunction;
    wrapper.instance().forceUpdate();
    wrapper.find("#test").simulate("mouseMove", event);
    expect(wrapper.text()).toEqual("Coordinates: (40, 40)");
  });
});
