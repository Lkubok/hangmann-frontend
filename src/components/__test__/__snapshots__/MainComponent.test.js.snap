// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box Component Should match snapshot after mount 1`] = `
<App
  changeFilter={[MockFunction]}
  changePageSize={[MockFunction]}
  changeSortedQuotes={[MockFunction]}
  filterQuery={null}
  pageSize={null}
  quotes={null}
>
  <div
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <LogicComponent
      render={[Function]}
    >
      <div
        onMouseMove={[Function]}
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <h1
          id="test"
        >
          Coordinates: (
          0
          , 
          0
          )
        </h1>
      </div>
    </LogicComponent>
  </div>
</App>
`;
